src/libsync/propagatorjobs.cpp

```diff
@@ -249,15 +258,37 @@ void PropagateLocalRename::start()
     }
 
     SyncJournalFileRecord oldRecord;
-    if (!propagator()->_journal->getFileRecord(_item->_originalFile, &oldRecord)) {
+    if (!propagator()->_journal->getFileRecord(fileAlreadyMoved ? previousNameInDb : _item->_originalFile, &oldRecord)) {
         qCWarning(lcPropagateLocalRename) << "could not get file from local DB" << _item->_originalFile;
         done(SyncFileItem::NormalError, tr("could not get file %1 from local DB").arg(_item->_originalFile));
         return;
     }
-    if (!propagator()->_journal->deleteFileRecord(_item->_originalFile)) {
-        qCWarning(lcPropagateLocalRename) << "could not delete file from local DB" << _item->_originalFile;
-        done(SyncFileItem::NormalError, tr("Could not delete file record %1 from local DB").arg(_item->_originalFile));
-        return;
+
+    const auto deleteOldRecord = [this] (const QString &fileName) -> bool
+    {
+        SyncJournalFileRecord oldRecord;
+        if (!propagator()->_journal->getFileRecord(fileName, &oldRecord)) {
+            qCWarning(lcPropagateLocalRename) << "could not get file from local DB" << fileName;
+            done(SyncFileItem::NormalError, tr("could not get file %1 from local DB").arg(fileName));
+            return false;
+        }
+        if (!propagator()->_journal->deleteFileRecord(fileName)) {
+            qCWarning(lcPropagateLocalRename) << "could not delete file from local DB" << fileName;
+            done(SyncFileItem::NormalError, tr("Could not delete file record %1 from local DB").arg(fileName));
+            return false;
+        }
+
+        return true;
+    };
```

Comment:
nitpick: I think it would make PropagateLocalRename::start more readable to make this a separate private method